plugins/inputs/logfile/logfile_test.go

```diff
@@ -406,49 +411,38 @@ func getLogSrc(t *testing.T, logFile *LogFile) (*logs.LogSrc, chan logs.LogEvent
 	return &logSource, evts
 }
 
-func writeLines(t *testing.T, file *os.File, numLines int, msg string) {
-	t.Log("Fill temp file with sufficient lines to be read.")
+func writeLines(t *testing.T, f *os.File, numLines int, msg string) {
+	t.Logf("start writing, %s", f.Name())
 	for i := 0; i < numLines; i++ {
-		_, err := file.WriteString(msg + "\n")
+		_, err := f.WriteString(msg + "\n")
 		require.NoError(t, err)
 	}
+	t.Logf("stop writing, %s", f.Name())
 }
 
 // createWriteRead creates a temp file, writes to it, then verifies events
 // are received. If isParent is true, then spawn a 2nd goroutine for createWriteRead.
-// Close the given channel when complete to let caller know it was successful.
+// Closes "done" channel when complete to let caller know it was successful.
 func createWriteRead(t *testing.T, prefix string, logFile *LogFile, done chan bool, isParent bool) {
 	// Let caller know when the goroutine is done.
 	defer close(done)
 	// done2 is only passed to child if this is the parent.
 	done2 := make(chan bool)
 	file := makeTempFile(t, prefix)
-	if isParent {
-		logFile = setupLogFileForTest(t, file, prefix)
-		defer logFile.Stop()
-	}
```

Comment:
Moved `setupLogFileForTest()` to the calling Function - `TestLogsFileAutoRemoval()`